File: salonmanager/backend/app/Models/Shift.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Support\Tenancy\SalonOwned;

class Shift extends Model
{
    use SalonOwned;

    protected $fillable = [
        'salon_id', 'stylist_id', 'user_id', 'start_at', 'end_at', 'status', 'meta',
        'role', 'title', 'rrule', 'exdates', 'published'
    ];
    
    protected $casts = [
        'start_at' => 'datetime', 
        'end_at' => 'datetime', 
        'meta' => 'array',
        'exdates' => 'array',
        'published' => 'boolean'
    ];

    public function stylist()
    {
        return $this->belongsTo(Stylist::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: salonmanager/frontend/lib/features/staff/models/shift.dart
class Shift {
  final int id;
  final int stylistId;
  final DateTime startAt;
  final DateTime endAt;
  final String status;
  final Map<String, dynamic> meta;

  Shift({
    required this.id,
    required this.stylistId,
    required this.startAt,
    required this.endAt,
    required this.status,
    this.meta = const {},
  });

  factory Shift.fromJson(Map<String, dynamic> j) => Shift(
        id: j['id'] as int,
        stylistId: j['stylist_id'] as int,
        startAt: DateTime.parse(j['start_at'] as String),
        endAt: DateTime.parse(j['end_at'] as String),
        status: j['status'] as String? ?? 'planned',
        meta: Map<String, dynamic>.from(j['meta'] ?? {}),
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'stylist_id': stylistId,
        'start_at': startAt.toIso8601String(),
        'end_at': endAt.toIso8601String(),
        'status': status,
        'meta': meta,
      };
}